frontend/src/components/pages/shadowlinks/details/shadow-link-details.tsx

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright 2025 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file https://github.com/redpanda-data/redpanda/blob/dev/licenses/bsl.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+'use client';
+
+import type { ShadowLink } from 'protogen/redpanda/api/console/v1alpha1/shadowlink_pb';
+import { useMemo, useState } from 'react';
+import { useListShadowTopicInfiniteQuery } from 'react-query/api/shadowlink';
+import { MAX_PAGE_SIZE } from 'react-query/react-query.utils';
+
+import { ShadowLinkDiagram } from './shadow-link-diagram';
+import { ShadowLinkMetrics } from './shadow-link-metrics';
+import { ShadowTopicsTable } from './shadow-topics-table';
+
+type ShadowLinkDetailsProps = {
+  shadowLink: ShadowLink;
+  shadowLinkName: string;
+  onFailoverTopic: (topicName?: string) => void;
+};
+
+export const ShadowLinkDetails = ({ shadowLink, shadowLinkName, onFailoverTopic }: ShadowLinkDetailsProps) => {
+  const [topicNameFilter, setTopicNameFilter] = useState('');
+
+  const {
+    data: shadowTopicsData,
+    refetch: refetchTopics,
+    fetchNextPage,
+    hasNextPage,
+    isFetchingNextPage,
+    isFetching: isFetchingTopics,
+  } = useListShadowTopicInfiniteQuery(
+    {
+      shadowLinkName,
+      filter: topicNameFilter
+        ? {
+            topicNameContains: topicNameFilter,
+          }
+        : undefined,
+      pageSize: MAX_PAGE_SIZE,
+    },
+    { refetchInterval: 15_000 }
```

Comment:
Let's use `SHORT_LIVED_CACHE_STALE_TIME`. That way in the future we can change it easily for all queries to be consistent.